drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -672,3 +674,8 @@ NET_DEVICE_DT_INST_DEFINE(0,
 		&esp32_data, NULL, CONFIG_WIFI_INIT_PRIORITY,
 		&esp32_api, ETHERNET_L2,
 		NET_L2_GET_CTX_TYPE(ETHERNET_L2), NET_ETH_MTU);
+
+#if defined(CONFIG_NET_CONNECTION_MANAGER_CONNECTIVITY_WIFI_MGMT)
+CONN_MGR_CONN_DECLARE_PUBLIC(CONNECTIVITY_WIFI_MGMT);
```

Comment:
I think it'd make sense to use `CONN_MGR_CONN_DECLARE_PUBLIC()` in a public header (`connectivity_wifi_mgmt.h`) where the context type is defined (as suggested in [docs](https://docs.zephyrproject.org/latest/connectivity/networking/conn_mgr/implementation.html#declaring-an-implementation-publicly)).

Reply:
Resolved as part of @jukkar's suggestion, although not in this way.